fedimint-core/src/util/mod.rs

```diff
@@ -307,13 +309,57 @@ pub fn handle_version_hash_command(version_hash: &str) {
     }
 }
 
+/// Run the supplied closure `op_fn` up to `max_attempts` times. Wait for the
+/// supplied `Duration` `interval` between attempts
+///
+/// # Returns
+///
+/// - If the closure runs successfully, the result is immediately returned
+/// - If the closure did not run successfully for `max_attempts` times, the
+///   error of the closure is returned
+pub async fn retry<F, Fut, T>(
+    op_name: impl Into<String>,
+    op_fn: F,
+    interval: Duration,
+    max_attempts: u32,
+) -> Result<T, anyhow::Error>
+where
+    F: Fn() -> Fut,
+    Fut: Future<Output = Result<T, anyhow::Error>>,
+{
+    let op_name = op_name.into();
+    assert_ne!(max_attempts, 0, "max_attempts must be greater than 0");
+    let mut attempts = 0;
+    loop {
+        attempts += 1;
+        match op_fn().await {
+            Ok(result) => return Ok(result),
+            Err(err) if attempts < max_attempts => {
+                // run closure op_fn again
+                info!(
```

Comment:
nit: `info` seems like a weird log level, given that we are still retrying I'd go with `debug` and only `warn` when actually failed.